src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -15001,35 +15001,204 @@ def stack(
+            for i in range(len(index_cols))
+        ]
+        # Set the correct index names based on column and index names
+        if self.columns.name is None and not has_index_names_none:
+            qc = qc.set_index_names(index_cols_replace_none + [None])  # type: ignore
+        elif self.columns.name is None and has_index_names_none:
+            qc = qc.set_index_names([None, None])
+        elif has_index_names_none:
+            qc = qc.set_index_names([col_label, None])
+
         if dropna:
             return qc.dropna(axis=0, how="any", thresh=None)
         else:
             return qc
 
+    def unstack(
+        self,
+        level: Union[int, str, list] = -1,
+        fill_value: Optional[Union[int, str, dict]] = None,
+        sort: bool = True,
+        is_series_output: bool = False,
+    ) -> "SnowflakeQueryCompiler":
+        """
+        Pivot a level of the (necessarily hierarchical) index labels.
+
+        Returns a DataFrame having a new level of column labels whose
+        inner-most level consists of the pivoted index labels.
+
+        If the index is not a MultiIndex, the output will be a Series
+        (the analogue of stack when the columns are not a MultiIndex).
+
+        Parameters
+        ----------
+        level : int, str, list, default -1
+            Level(s) of index to unstack, can pass level name.
+
+        fillna : int, str, dict, optional
+            Replace NaN with this value if the unstack produces missing values.
+
+        sort : bool, default True
+            Sort the level(s) in the resulting MultiIndex columns.
+
+        is_series_output : bool, default False
+            Whether the output is a Series, used by `fillna`
+        """
+        if not isinstance(level, int):
+            ErrorMessage.not_implemented(
+                "Snowpark pandas DataFrame/Series.unstack does not yet support a non-integer `level` parameter"
+            )
+        if not sort:
+            ErrorMessage.not_implemented(
+                "Snowpark pandas DataFrame/Series.unstack does not yet support the `sort` parameter"
+            )
+        level = level if is_list_like(level) else [level]
+
+        index_names = self.get_index_names()
+        has_index_names_none = None in index_names
+        # Check to see if we have a MultiIndex, if we do, make sure we remove
+        # the appropriate level(s), and we pivot accordingly.
+        if len(index_names) > 1:
+            if has_index_names_none:
+                # Index name defaults to "level_{i}" after reset_index() operation
+                # if index name is None
+                index_cols = [
+                    f"level_{i}" if index_names[i] is None else index_names[i]
+                    for i in range(len(index_names))
+                ]
+            else:
+                index_cols = index_names  # type: ignore
+
+            pivot_cols = [index_cols[lev] for lev in level]  # type: ignore
```

Comment:
This still works for level -1 (see the tests), it will simply grab the last column in index_cols. For a two-level MultiIndex, -1 has the same behavior as level 1.